chalk/trail.py

```diff
@@ -1,110 +1,166 @@
 from __future__ import annotations
 
 from dataclasses import dataclass
-from typing import List
+from functools import reduce
+from typing import TYPE_CHECKING, Iterable, List, Tuple, Union
 
-from chalk.shapes.path import Path
+from chalk.envelope import Envelope
+from chalk.shapes.arc import ArcSegment, arc_seg_angle
+from chalk.shapes.segment import Segment, seg
+from chalk.trace import Trace
 from chalk.transform import (
     P2,
     V2,
     Affine,
     Transformable,
-    Vec2Array,
     apply_affine,
+    remove_linear,
     remove_translation,
+    unit_x,
+    unit_y,
 )
-from chalk.types import Diagram
+from chalk.types import Diagram, Enveloped, Traceable, TrailLike
+
+if TYPE_CHECKING:
+    from chalk.shapes.path import Path
+
+SegmentLike = Union[Segment, ArcSegment]
 
 
 @dataclass
-class Trail(Transformable):
-    """Trail class.
+class Located(Enveloped, Traceable, Transformable):
+    trail: Trail
+    location: P2
 
-    This is derived from a ``chalk.transform.Transformable`` class.
+    def located_segments(self) -> Iterable[Tuple[SegmentLike, P2]]:
+        return zip(self.trail.segments, self.points())
 
-    [TODO]: Need more explanation on what this class is for (preferably
-    with illustrations/figures).
-    """
+    def points(self) -> Iterable[P2]:
+        return (pt + self.location for pt in self.trail.points())
 
-    offsets: List[V2]
+    def get_envelope(self) -> Envelope:
+        return Envelope.concat(
+            segment.get_envelope().translate_by(location)
+            for segment, location in self.located_segments()
+        )
+
+    def get_trace(self) -> Trace:
+        return Trace.concat(
+            segment.get_trace().translate_by(location)
+            for segment, location in self.located_segments()
+        )
+
+    def stroke(self) -> Diagram:
+        return self.to_path().stroke()
+
+    def apply_transform(self, t: Affine) -> Located:  # type: ignore
```

Comment:
Shouldn't the code here rather be:
```diff
     def apply_transform(self, t: Affine) -> Located:  # type: ignore
-        t2 = remove_linear(t)
+        t2 = remove_translation(t)
         return Located(
-            apply_affine(t, self.trail), apply_affine(t2, self.location)
+            apply_affine(t2, self.trail), apply_affine(t, self.location)
         )
```
I haven't tested this thoroughly, but this change was needed for [the example in the first comment](https://github.com/danoneata/chalk/pull/102#issuecomment-1214242619) (`dia2`).

Reply:
Nice catch. Fixed.